client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -381,6 +383,17 @@ static Request clearScroll(ClearScrollRequest clearScrollRequest) throws IOExcep
         return new Request("DELETE", "/_search/scroll", Collections.emptyMap(), entity);
     }
 
+    static Request multiSearch(MultiSearchRequest multiSearchRequest) throws IOException {
+        Params params = Params.builder();
+        if (multiSearchRequest.maxConcurrentSearchRequests() != 0) {
```

Comment:
Not sure why I did this... I'll always pass down this parameter.

Reply:
I remember now why I needed to add this. `0` is used as not specified and the `maxConcurrentSearchRequests(...)` setter doesn't accepts values lower than 1. So checking whether any other value than 0 is specified is required.

Reply:
I see, yea we can't avoid this then. Maybe share the default value through a constant so at least we don't duplicate it. Odd! :)

Reply:
this is odd especially because it seems that once you set a value for this field, you can never reset it to its original default value.

Reply:
agreed, I think this should be fixed and the setter should allow `0` as valid value. So that the msearch api falls back to default based on number of cores.